File: CodingInterviewsPythonSolutions/Subarray Sum Equals K.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        myMap = {}
        myMap[0] = 1
        sz = len(nums)
        result = 0
        prefixSum = 0
        for i in range(sz):
            prefixSum += nums[i]
            if prefixSum - k in myMap:
                result += myMap[prefixSum - k]
            myMap[prefixSum] = myMap.get(prefixSum, 0) + 1
        return result

File: CodingInterviewsPackSolutionsCPP/Subarray Sum Equals K.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int>myMap;
        myMap[0] = 1;
        int sz = nums.size();
        int result = 0;
        int prefixSum = 0;
        for(int i = 0; i < sz; i++){
            prefixSum+=nums[i];
            if(myMap.count(prefixSum-k))
                result+=myMap[prefixSum-k];
            
            myMap[prefixSum]++;
        }
        return result;
    }
};
